src/pathfora.js

```diff
@@ -1895,131 +1901,143 @@
-          var requestGoogle = false;
+    /**
+     * @description Load callback for google integration
+     * @param {object} optional widget object
+     */
+    onGoogleLoad: function() {
+      gapi.load('auth2', function() {
+        var auth2 = gapi.auth2.init({
+          clientId: pathforaDataObject.socialNetworks.googleClientID,
+          cookiepolicy: 'single_host_origin',
+          scope: 'profile'
+        });
 
-          if (typeof element.querySelector === 'undefined') {
-            return false;
-          }
+        var googleBtns = Array.prototype.slice.call(document.querySelectorAll('.social-login-btn.google-login-btn span'));
 
-          requestFacebook = element.querySelector('.fb-login-button') !== null;
-          requestGoogle = element.querySelector('.google-login') !== null;
+        auth2.then(function() {
+          var user = auth2.currentUser.get();
+          core.autoCompleteGoogleData(user, googleBtns);
 
-          if (requestFacebook) {
-            renderWidgets.facebook = renderWidgets.facebook.concat(element);
-          }
-          if (requestGoogle) {
-            renderWidgets.google = renderWidgets.google.concat(element);
-          }
+          googleBtns.forEach(function(element) {
+            if (element.parentElement) {
+              element.parentElement.onclick = function() {
+                core.onGoogleClick(googleBtns);
+              }
+            }
+          });
         });
-
-        timeoutInterval['fb'] = attemptRenderingFacebook();
-        timeoutInterval['google'] = attemptRenderingGoogle();
-      };
-
-      this.requestSocialPluginRender(arguments[0]);
+      });
     },
 
     /**
-     * @description Attempt to load forms' data from Facebook API.
-     * @throws {Error} Facebook API Error
+     * @description Attempt to load forms' data from Google+ API.
+     * @param {object} current googleUser
+     * @param {object} google buttons element selector
      */
-    autoCompleteFacebookData: function () {
-      if (typeof window.FB !== 'undefined') {
-        window.FB.getLoginStatus(function (connection) {
-          if (connection.status === 'connected') {
-            window.FB.api('/me', {
-              fields: 'name,first_name,last_name,email'
-            }, function (query) {
-              if (query.error) {
-                throw new Error('Facebook API Error: ' + query.error);
-              }
+    autoCompleteGoogleData: function (user, elements) {
+      if (typeof user !== 'undefined') {
+        var profile = user.getBasicProfile();
 
-              core.autoCompleteFormFields({
-                username: query.name || '',
-                email: query.email || '',
-                firstName: query.first_name || '',
-                lastName: query.last_name || ''
-              });
-            });
-          }
-        });
+        if (typeof profile !== 'undefined') {
+          core.autoCompleteFormFields({
+            type: "google",
```

Comment:
total nit but looks like we use single quotes in most places but double here (linter will catch this stuff also if you are in the process of adding that)